src/v/cluster/partition.cc

```diff
@@ -1165,6 +1165,100 @@ ss::future<> partition::unsafe_reset_remote_partition_manifest(iobuf buf) {
       ntp());
 }
 
+ss::future<> partition::unsafe_reset_remote_partition_manifest_from_cloud() {
+    vlog(
+      clusterlog.info,
+      "[{}] Unsafe manifest reset from cloud state requested",
+      ntp());
+
+    if (!(config::shard_local_cfg().cloud_storage_enabled()
+          && _archival_meta_stm)) {
+        vlog(
+          clusterlog.warn,
+          "[{}] Archival STM not present. Skipping unsafe reset ...",
+          ntp());
+        throw std::runtime_error("Archival STM not present");
+    }
+
+    // Pause the archiver to ensure that the reset operation is safe
+    auto paused = co_await _archiver->pause(20s);
+    auto unpause_guard = ss::defer([this] {
+        if (_archiver) {
+            _archiver->unpause();
+        }
+    });
+
+    if (!paused) {
+        vlog(clusterlog.warn, "[{}] Failed to pause remote write path", ntp());
+        throw std::runtime_error("Failed to pause remote write path");
+    }
+
+    // Download the current partition manifest from the cloud
+    cloud_storage::partition_manifest new_manifest{
+      _archiver->get_ntp(), _archiver->get_revision_id()};
+
+    auto timeout
+      = config::shard_local_cfg().cloud_storage_manifest_upload_timeout_ms();
+    auto backoff = config::shard_local_cfg().cloud_storage_initial_backoff_ms();
+
+    retry_chain_node rtc(_as, timeout, backoff);
+    auto [res, res_fmt]
+      = co_await _cloud_storage_api.local().try_download_partition_manifest(
+        _archiver->get_bucket_name(), new_manifest, rtc);
+
+    if (res != cloud_storage::download_result::success) {
+        throw std::runtime_error(ssx::sformat(
+          "Failed to download partition manifest with error: {}", res));
+    }
+
+    if (new_manifest.get_last_offset() < _raft->start_offset()) {
+        throw std::runtime_error(ssx::sformat(
+          "Applying the cloud manifest would cause data loss since the last "
+          "offset in the downloaded manifest is below the start offset of the "
+          "local log: {} < {}",
+          new_manifest.get_last_offset(),
+          _raft->start_offset()));
+    }
+
+    // Replicate the reset command which contains the downloaded manifest
+    auto replication_deadline = ss::lowres_clock::now() + 20s;
+    std::vector<cluster::command_batch_builder> builders;
+
+    auto reset_builder = _archival_meta_stm->batch_start(
+      replication_deadline, _as);
+    reset_builder.replace_manifest(new_manifest.to_iobuf());
+
+    vlog(
+      clusterlog.info,
+      "Replicating replace manifest command. New manifest details: {{ "
+      "start_offset: {}, last_offset: {}}}",
+      new_manifest.get_start_offset(),
+      new_manifest.get_last_offset());
+
+    auto errc = co_await reset_builder.replicate();
```

Comment:
are we guaranteed to still be a leader in the same term as when we requested the archival pause? or it could have flopped to another node and potentially back?

Reply:
If the node is not a leader anymore this 'replicate' call with fail.